dynamodb_python/table.py

```diff
@@ -0,0 +1,428 @@
+
+        if limit or limit == 0:
+            limit = min(max(0, limit), 200)
+            if limit == 0:
+                return [], {}
+            kwargs["Limit"] = limit * 2
+
+        if last:
+            kwargs["ExclusiveStartKey"] = self._serialize(
+                (
+                    {
+                        "key": last["key"],
+                        "time": int(last["time"]),
+                    }
+                )
+            )
+
+        items = []
+        last_key = {}
+        parse_more = True
+
+        while parse_more:
+            response = self._client.query(TableName=self.tablename, **kwargs)
+            items.extend(self._deserialize(response["Items"]))
+            last_key = response.get("LastEvaluatedKey", {})
+            parse_more = ((limit and len(items) < limit) or limit is None) and last_key
+            kwargs["ExclusiveStartKey"] = last_key
+
+        if limit and len(items) > limit:
+            truncated_items = items[:limit]
+            last_key = truncated_items[-1]
+            return truncated_items, last_key
+        return items, self._deserialize(last_key)
+
+    def delete(self, key: str, time: int) -> dict:
+        """Delete item and return its values."""
+        response = self._client.delete_item(
+            TableName=self.tablename,
+            Key=self._serialize(({"key": key, "time": time})),
+            ReturnValues="ALL_OLD",
+        )
+        if "Attributes" not in response:
+            raise KeyError("Key(%s) and time(%s) was not found." % (key, time))
+        return self._deserialize(response["Attributes"])
+
+    def delete_list(self, keys: list, time: int) -> dict:
+        """Delete same "time" from multiple keys."""
+        response = {}
+        for key in keys:
+            response = self.delete(key, time)
+        return response
+
+    def write_list(self, keys: list, time: int, data: dict) -> dict:
+        """Write same data with same "time" to multiple keys."""
+        response = {}
+        for key in keys:
+            response = self.write(key, time, data)
+        return response
+
+    def exists(self, key: str, time: int) -> bool:
+        """Check whether item with such key and "time" exists."""
+        try:
+            self.read_item(key, time)
+            return True
+        except KeyError:
+            return False
+
+    def write(
+        self,
+        key: str,
+        time: int,
+        data: dict,
+        return_old_values: bool = False,
+        ttl: datetime = None,
+    ) -> dict:
+        """Writing to DynamoDB database"""
+        item = {**data, "key": key, "time": time}
+
+        if ttl:
+            item["ttl"] = int(ttl.timestamp())
+
+        return_values = "ALL_OLD" if return_old_values else "NONE"
+
+        # update updated_by. updated_at fields before writing.
+        if "updated_by" in item:
+            try:
+                data["updated_by"] = item["updated_by"] = AUTH_NAME
```

Comment:
If this is going to be open sourced - I suppose we need to remove this thing, it is our internal thing.